src/fp.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2021 The Authors
+// SPDX-License-Identifier: MPL-2.0
+
+use rand::prelude::*;
+use rand::Rng;
+
+// Parameters of a finite field with prime p < 2^126. Field elements are represented as u128's.
+#[derive(Debug)]
+pub struct Fp {
```

Comment:
What does `Fp` stand for? Can we use a more descriptive module and struct name than `Fp`?

Reply:
"Field parameters". How about `FieldParam`?

Reply:
Why not `FieldParameters`? I think we can afford the extra five characters.

Reply:
Works for me.